File: MIDIC#/MIDILibrary/MIDILibrary/HeaderChunk.cs
namespace MIDILibrary
{
    public class HeaderChunk : IChunk
    {
        private readonly uint _length;
        private readonly ushort _format;
        public ushort TrackCount { get; }
        public ushort TickDivision { get; }

        public HeaderChunk(uint length, ushort format, ushort trackCount, ushort tickDivision)
        {
            _length = length;
            _format = format;
            TrackCount = trackCount;
            TickDivision = tickDivision;
        }

        public uint GetLength()
        {
            return 8 + _length;
        }

        public byte[] ToBitString()
        {
            var bitString = new byte[GetLength()];
            uint cursor = 0;
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, MidiConstants.MIDI_SIGNATURE);
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, _length);
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, _format);
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, TrackCount);
            StreamingHelper.StreamValueToBitString(ref cursor, ref bitString, TickDivision);
            return bitString;
        }
    }
}

File: MIDIJava/src/io/github/seanboyy/Midi/HeaderChunk.java
package io.github.seanboyy.Midi;

public class HeaderChunk implements Chunk {
    private int length;
    private short format;
    private short trackCount;
    private short tickDivision;

    public short getTickDivision() {
        return tickDivision;
    }

    short getTrackCount() {
        return trackCount;
    }

    HeaderChunk(int length, short format, short trackCount, short tickDivision) {
        this.length = length;
        this.format = format;
        this.trackCount = trackCount;
        this.tickDivision = tickDivision;
    }

    @Override
    public byte[] toBitString() {
        byte[] bitString = new byte[getLength()];
        int[] cursor = {0};
        bitString = StreamingHelper.streamValueToBitString(cursor, bitString, MidiConstants.MIDI_SIGNATURE);
        bitString = StreamingHelper.streamValueToBitString(cursor, bitString, length);
        bitString = StreamingHelper.streamValueToBitString(cursor, bitString, format);
        bitString = StreamingHelper.streamValueToBitString(cursor, bitString, trackCount);
        bitString = StreamingHelper.streamValueToBitString(cursor, bitString, tickDivision);
        return bitString;
    }

    @Override
    public int getLength() {
        return 8 + length;
    }
}
